tests/all/manifest.rs

```diff
@@ -481,3 +482,46 @@ fn it_lists_license_files_in_files_field_of_package_json() {
         pkg.files,
     );
 }
+
+#[test]
+fn it_recurses_up_the_path_to_find_cargo_toml() {
+    let fixture = utils::fixture::Fixture::new();
+    fixture.hello_world_src_lib().file(
+        "Cargo.toml",
+        r#"
+            [package]
+            authors = ["The wasm-pack developers"]
+            description = "so awesome rust+wasm package"
+            license = "WTFPL"
+            name = "recurse-for-manifest-test"
+            repository = "https://github.com/rustwasm/wasm-pack.git"
+            version = "0.1.0"
+            homepage = "https://rustwasm.github.io/wasm-pack/"
+        "#,
+    );
+    let path = get_crate_path(None).unwrap();
+    let crate_data = manifest::CrateData::new(&path, None).unwrap();
+    let name = crate_data.crate_name();
+    assert_eq!(name, "wasm_pack");
+}
+
+#[test]
+fn it_doesnt_recurse_up_the_path_to_find_cargo_toml_when_default() {
+    let fixture = utils::fixture::Fixture::new();
+    fixture.hello_world_src_lib().file(
+        "Cargo.toml",
+        r#"
+            [package]
+            authors = ["The wasm-pack developers"]
+            description = "so awesome rust+wasm package"
+            license = "WTFPL"
+            name = "recurse-for-manifest-test"
+            repository = "https://github.com/rustwasm/wasm-pack.git"
+            version = "0.1.0"
+            homepage = "https://rustwasm.github.io/wasm-pack/"
+        "#,
+    );
+    let path = get_crate_path(Some(PathBuf::from("src/"))).unwrap();
```

Comment:
```suggestion
    let path = get_crate_path(Some(PathBuf::from("src"))).unwrap();
```

Let's avoid hard coding path separators.